plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.test.ts

```diff
@@ -156,17 +185,22 @@ describe('JsonRulesEngineFactChecker', () => {
         'Failed to run rules engine, Undefined fact: somefact',
       );
     });
-    it('should respond with result, facts, fact schemas and checks', async () => {
-      latestFactsByIdsMock.mockImplementation(() =>
-        Promise.resolve({
-          ['test-factretriever']: {
-            id: 'test-factretriever',
-            facts: {
-              testnumberfact: 3,
-            },
-          },
+
+    it('should skip checks where fact data is missing', async () => {
+      const skipped = async () =>
+        await factChecker.runChecks('a/a/a', ['brokennotfound']);
+      await expect(skipped()).resolves.toEqual([]);
+
+      const partial = async () =>
+        await factChecker.runChecks('a/a/a', ['brokennotfound', 'simple']);
+      await expect(partial()).resolves.toEqual([
+        expect.objectContaining({
+          check: expect.objectContaining({ id: 'simpleTestCheck' }),
         }),
-      );
+      ]);
```

Comment:
Is this potentially a seperate test case?

Reply:
We can split it up if you prefer. I just figured there should be a check to make sure we still get the valid ones, even if some are bad.

Reply:
Would it make sense as just one test case where you check that only one fact check result is returned for `simple`?

Reply:
I'll let you decide as it doesn't really matter 🙂